### PR TITLE
WebDriver: Some WebDriver session management fixes

### PR DESCRIPTION
**Pull request by trflynn89**

### CODE REVIEW

File: Userland/Services/WebContent/WebDriverConnection.cpp [hunk 1]
```diff
@@ -1745,11 +1745,18 @@ Messages::WebDriverClient::PrintPageResponse WebDriverConnection::print_page()
 }
 
 // https://w3c.github.io/webdriver/#dfn-no-longer-open
-ErrorOr<void, Web::WebDriver::Error> WebDriverConnection::ensure_open_top_level_browsing_context()
+Messages::WebDriverClient::EnsureTopLevelBrowsingContextIsOpenResponse WebDriverConnection::ensure_top_level_browsing_context_is_open()
 {
     // A browsing context is said to be no longer open if it has been discarded.
     if (m_page_client.page().top_level_browsing_context().has_been_discarded())
         return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::NoSuchWindow, "Window not found"sv);
```
Comment (kalenikaliaksandr) [inline Userland/Services/WebContent/WebDriverConnection.cpp:1752]: hmm, I think that "1. If the current top-level browsing context is no longer open, return error with error code no such window." step doesn't make a lot of sense on web content side and should rather be a check if connection is with a required window is present on web driver process side.

Comment (trflynn89) [reply Userland/Services/WebContent/WebDriverConnection.cpp:1752]: I don't think that is sufficient. For example, consider calling `window.close()` from JavaScript - the browsing context will be discarded, and if we don't ask the WebContent process if that happened, we'll incorrectly not throw an error here.

Comment (kalenikaliaksandr) [reply Userland/Services/WebContent/WebDriverConnection.cpp:1752]: yeah, agree, unless WebContent will let WebDriver process know that it's going to be destroyed this check has its place.

Comment (trflynn89) [reply Userland/Services/WebContent/WebDriverConnection.cpp:1752]: Yeah, we'll need to come up with a general solution for keeping the local and remote endpoints in sync.

**Review by linusg — APPROVED** [review]

**Pull request merged**